### ISSUE TITLE
Trandformer model performance

### ISSUE THREAD

**Issue by zaynmi** [KaihuaTang/Scene-Graph-Benchmark.pytorch#22]
## ❓ Questions and Help
I used the default configs except the SOLVER.BASE_LR set to 0.001 to train the Transformer model.  However, I couldn't get the reported results. The R@K drop by ~2% for three protocols. 

What other configs should I change to train the Transformer model to reach the reported results?

Thanks

**Comment by KaihuaTang** [comment]
Another issue reported the same problem: https://github.com/KaihuaTang/Scene-Graph-Benchmark.pytorch/issues/21

My results are all from the default settings, except those with special announcements, which means use the default SOLVER.BASE_LR may help you get my results.

**Comment by zaynmi** [comment]
@KaihuaTang Hi, I tried default SOLVER.BASE_LR 0.01 and 0.005, but the loss got Nan.

**Comment by KaihuaTang** [comment]
> @KaihuaTang Hi, I tried default SOLVER.BASE_LR 0.01 and 0.005, but the loss got Nan.

How about the batch_size? Did you use the SOLVER.IMS_PER_BATCH 12 ?

**Comment by zaynmi** [comment]
Yes, the batch size is 12 and I used the command: (other default configs are not changed)
```
CUDA_VISIBLE_DEVICES=2,3 python -m torch.distributed.launch --master_port 10025 --nproc_per_node=2 \
tools/relation_train_net.py --config-file "configs/e2e_relation_X_101_32_8_FPN_1x.yaml" \
MODEL.ROI_RELATION_HEAD.USE_GT_BOX True MODEL.ROI_RELATION_HEAD.USE_GT_OBJECT_LABEL False \
MODEL.ROI_RELATION_HEAD.PREDICTOR TransformerPredictor SOLVER.IMS_PER_BATCH 12 TEST.IMS_PER_BATCH 2 DTYPE "float16" \
SOLVER.MAX_ITER 50000 SOLVER.VAL_PERIOD 5000 SOLVER.CHECKPOINT_PERIOD 5000 GLOVE_DIR /home/alice/glove \
OUTPUT_DIR /data2/alice/sgg/transformer-sgcls-exmp SOLVER.BASE_LR 0.005
```

**Comment by KaihuaTang** [comment]
> Yes, the batch size is 12 and I used the command: (other default configs are not changed)
> 
> ```
> CUDA_VISIBLE_DEVICES=2,3 python -m torch.distributed.launch --master_port 10025 --nproc_per_node=2 \
> tools/relation_train_net.py --config-file "configs/e2e_relation_X_101_32_8_FPN_1x.yaml" \
> MODEL.ROI_RELATION_HEAD.USE_GT_BOX True MODEL.ROI_RELATION_HEAD.USE_GT_OBJECT_LABEL False \
> MODEL.ROI_RELATION_HEAD.PREDICTOR TransformerPredictor SOLVER.IMS_PER_BATCH 12 TEST.IMS_PER_BATCH 2 DTYPE "float16" \
> SOLVER.MAX_ITER 50000 SOLVER.VAL_PERIOD 5000 SOLVER.CHECKPOINT_PERIOD 5000 GLOVE_DIR /home/alice/glove \
> OUTPUT_DIR /data2/alice/sgg/transformer-sgcls-exmp SOLVER.BASE_LR 0.005
> ```

Pls allow me to check

**Comment by KaihuaTang** [comment]
The given hyper-parameters indeed fail to reproduce the results of Transformer. (Since my cvpr 2020 paper didn't include this model, I lost the original checkpoints and the parameters I used for  the results.) But I have figured out the main reasons: 
1) For transformer model, instead of using the SOLVER.SCHEDULE.TYPE = WarmupReduceLROnPlateau (in configs/e2e_relation_X_101_32_8_FPN_1x.yaml), I used a more conventional scheduler WarmupMultiStepLR. 
2) I may use an even larger number of batch_size. Since this transformer model is more memory-efficient, the maximum batch_size is 20 in my RTX 2080ti. I guess I use 18 or 20 for the transformer model.

However, I'm still trying to find out the optimum SOLVER.STEPS,  SOLVER.MAX_ITER and SOLVER.IMS_PER_BATCH that can reproduce the results.  I will let you know once I reproduce the reported results.

**Comment by zaynmi** [comment]
@KaihuaTang Thank you so much to help me.

**Comment by KaihuaTang** [comment]
It seems like using SOLVER.SCHEDULE.TYPE WarmupMultiStepLR, SOLVER.STEPS, SOLVER.MAX_ITER 16000, SOLVER.IMS_PER_BATCH 16, SOLVER.STEPS (10000, 16000)  helping me reproduce a very close results for PredCls.

![image](https://user-images.githubusercontent.com/24402685/79716628-d6358f00-8309-11ea-8109-482a7225348c.png)

**Comment by KaihuaTang** [comment]
I think the main difference is that the learning rate of Transformer model should only decay once (decay twice may get overfit)
